cpp/src/sort/rank.cu

```diff
@@ -160,15 +160,17 @@ void rank_dense(rmm::device_vector<size_type> const &dense_rank_sorted,
 }
 
 template <typename outputType>
-void rank_min(rmm::device_vector<size_type> const &group_keys,
+void rank_min(cudf::device_span<size_type const> group_keys,
               column_view sorted_order_view,
               mutable_column_view rank_mutable_view,
               rmm::cuda_stream_view stream)
 {
   // min of first in the group
   // All equal values have min of ranks among them.
   // algorithm: reduce_by_key(dense_rank, 1, n, min), scatter
+  rmm::device_uvector<size_type> tie_sorted(sorted_order_view.size(), stream);
```

Comment:
I'm curious - why move this here only to pass it back to the function that was formerly declaring it?  `tie_break_ranks_transform` takes the stream parameter so it can still do the declaration.

`rank_max` also does this, so now we have this declaration in two places instead of just the (original) one.

Reply:
because `device_uvector<thrust::tuple>` won't compile.

Reply:
Ah I see. I didn't realize rmm::device_vector<MinCount> remained as is.

Reply:
This is all simplified now thanks to @karthikeyann 's suggestion.